daft/io/_csv.py

```diff
@@ -45,28 +46,27 @@ def read_csv(
         path (str): Path to CSV (allows for wildcards)
         schema_hints (dict[str, DataType]): A mapping between column names and datatypes - passing this option
             will override the specified columns on the inferred schema with the specified DataTypes
+        schema_override (dict[str, DataType]): A definitive schema to use for the CSV, passing this option will
+            disable schema inference
         has_headers (bool): Whether the CSV has a header or not, defaults to True
         delimiter (Str): Delimiter used in the CSV, defaults to ","
         doubled_quote (bool): Whether to support double quote escapes, defaults to True
         escape_char (str): Character to use as the escape character for double quotes, or defaults to `"`
         comment (str): Character to treat as the start of a comment line, or None to not support comments
+        flexible (bool): Whether to allow for variable number of columns in the CSV, defaults to False
```

Comment:
We should include the exact semantics for "append nulls to rows with less columns than the schema, and ignore extra columns in rows with more columns" in the docstring.